solr/contrib/ltr/src/java/org/apache/solr/ltr/ranking/ModelQuery.java

```diff
@@ -152,12 +159,28 @@ public SolrQueryRequest getRequest() {
     return request;
   }
 
-  @Override  
-  public ModelWeight createWeight(IndexSearcher searcher, boolean needsScores, float boost) throws IOException {
+  @Override
+  public ModelWeight createWeight(IndexSearcher searcher, boolean needsScores, float boost)
+      throws IOException {   
+    long  start = System.currentTimeMillis();
+    
+    ModelWeight mw = null;
+    if(LTRThreadInterface.maxThreads <= 1){
```

Comment:
should consider also chacking # of features to create here.  If someone has multiple models deployed that rerank different categories of documents, one model might only have 5-10 features which might benefit from not going through the parallel approach.